### PR TITLE
Electrical optimisation and APC  Department battery connection

### PR DESCRIPTION
**Pull request by Bod9001**
### Purpose
Increase the performance of electrical system
and Make APC  Department battery connection Probably

### Open Questions and Pre-Merge TODOs

- [X]  This fix is tested on the same branch it is PR'ed to.
- [x]  I correctly commented my code
- [x]  My code is indented with tabs and not spaces
- [x]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [ ]  This PR has been tested in multiplayer

### Notes:
It still not quite finished the cable between the Jumps need to be updated the values

### CODE REVIEW

File: UnityProject/Assets/Scripts/Electricity/PoweredDevices/APC.cs [hunk 1]
```diff
@@ -19,6 +19,9 @@ public class APC : NetworkBehaviour, IElectricalNeedUpdate, IDeviceControl
 	/// <summary>
 	/// The current voltage of this APC. Calls OnVoltageChange when changed.
 	/// </summary>
+	/// 
+	public int CashOfConnectedDevices = 0;
```
Comment (Jibbajabbafic) [inline UnityProject/Assets/Scripts/Electricity/PoweredDevices/APC.cs:23]: Could you move this so it doesn't steal the documentation for the voltage

**Review by Jibbajabbafic — COMMENTED** [review]
There's a severe lack of comments overall. It might seem obvious to you what the code is doing, but anyone trying to work with this will have to spend a lot of time familiarising themselves with it. Some comments here and there explaining it would help speed that up a lot.

**Comment by Bod9001** [comment]
yeah,  Shall I add more comments as well as the Readme.md I have added

**Comment by Jibbajabbafic** [comment]
Yes please, cheers!

**Comment by chairbender** [comment]
sorry, my bad. typo in a pr. reopening

**Review by Jibbajabbafic — APPROVED** [review]
Looks good to me!

**Comment by DooblyNoobly** [comment]
does this PR satisfy any of the requirements for: #1442

**Comment by Jibbajabbafic** [comment]
I believe it does address some performance issues.
After some wrestling with deep profile you can see the results.

- This is before the PR:
before_pr_arrows

- This is after the PR (note how spikes are much smaller during idle):
after_pr_arrows

**Comment by DooblyNoobly** [comment]
Cool, will consider CPU optimization done but will leave GC criteria as there is still 400 - 500KB

**Pull request merged**